assets/js/modules/analytics/common/profile-select.test.js

```diff
@@ -0,0 +1,141 @@
+import * as fixtures from '../datastore/__fixtures__';
+import { fireEvent, render, act } from '../../../../../tests/js/test-utils';
+
+// Mock apiFetch so we know if it's called.
+jest.mock( '@wordpress/api-fetch' );
+apiFetchMock.mockImplementation( ( ...args ) => {
+	// eslint-disable-next-line no-console
+	console.warn( 'apiFetch', ...args );
+} );
+
+const setupRegistry = ( { dispatch } ) => {
+	const { id, webPropertyId, accountId } = fixtures.propertiesProfiles.profiles[ 0 ];
+	dispatch( modulesAnalyticsStoreName ).setAccountID( accountId );
+	dispatch( modulesAnalyticsStoreName ).setPropertyID( webPropertyId );
+	dispatch( modulesAnalyticsStoreName ).setProfileID( id );
+	dispatch( modulesAnalyticsStoreName ).receiveProfiles( fixtures.propertiesProfiles.profiles );
+};
+
+const setupRegistryWithExistingTag = ( { dispatch } ) => {
+	const existingTag = {
+		accountID: fixtures.accountsPropertiesProfiles.profiles[ 0 ].accountId,
+		propertyID: fixtures.accountsPropertiesProfiles.profiles[ 0 ].webPropertyId,
+	};
+	const { id } = fixtures.propertiesProfiles.profiles[ 0 ];
+	dispatch( modulesAnalyticsStoreName ).setAccountID( existingTag.accountID );
+	dispatch( modulesAnalyticsStoreName ).setPropertyID( existingTag.propertyID );
+	dispatch( modulesAnalyticsStoreName ).setProfileID( id );
+	dispatch( modulesAnalyticsStoreName ).receiveProfiles( fixtures.accountsPropertiesProfiles.profiles );
+	dispatch( modulesAnalyticsStoreName ).receiveExistingTag( existingTag.propertyID );
+};
+
+const setupEmptyRegistry = ( { dispatch } ) => {
+	dispatch( modulesAnalyticsStoreName ).setSettings( {} );
+	dispatch( modulesAnalyticsStoreName ).receiveProfiles( [] );
+};
+
+describe( 'ProfileSelect', () => {
+	afterEach( () => apiFetchMock.mockClear() );
+	afterAll( () => jest.restoreAllMocks() );
+
+	it( 'should render an option for each analytics profile of the currently selected account and property.', async () => {
+		const { getAllByRole } = render( <ProfileSelect />, { setupRegistry } );
+
+		const listItems = getAllByRole( 'menuitem', { hidden: true } );
+		// Note: we do length + 1 here because there should also be an item for
+		// "Set up a new property".
+		expect( listItems ).toHaveLength( fixtures.accountsPropertiesProfiles.profiles.length + 1 );
+	} );
+
+	it( 'should display profile options of an existing account when present, and not be disabled.', async () => {
+		const { container, getAllByRole, registry } = render( <ProfileSelect />, { setupRegistry: setupRegistryWithExistingTag } );
+
+		const currentPropertyID = registry.select( modulesAnalyticsStoreName ).getPropertyID();
+		const existingTagPropertyID = registry.select( modulesAnalyticsStoreName ).getExistingTag();
+		expect( existingTagPropertyID ).toEqual( currentPropertyID );
+
+		const existingTagProfiles = fixtures.accountsPropertiesProfiles.profiles
+			.filter( ( p ) => p.webPropertyId === existingTagPropertyID );
```

Comment:
Agreed. Still haven't gotten over the fact that this is recommended in Go 😄

Reply:
Actually, I meant to come back to these. The `p` isn't necessary at all since we can destructure its `webPropertyId` out of it, which is the only thing we care about in it.